### ISSUE TITLE
integer overflow?

### ISSUE THREAD

**Issue by lynxoid** [marbl/Mash#17]
Hi guys,

Ran `mash` on a collection of reference 16s rRNA sequences (3.6Gb) and got this warning while sketching:

```
WARNING: For the k-mer size used (21), the random match probability (0.000726951) is above 
the specified warning threshold (0.01) for the sequence "..rdp/current_Bacteria_unaligned.fa" of 
size 18446744072614073515. Distances to this sequence may be underestimated as a result. 
To meet the threshold of 0.01, a k-mer size of at least 20 is required.
```

Seems like the message should _not_ be displayed and/or an interger overflow happened when calculating the size of the reference.

My suspicion is that maybe there an assumption that the reference fasta will always contain a single sequence?

**Comment by aphillippy** [comment]
Hey Darya, yes, that genome size looks completely wonky. We'll take a look. Thanks!

**Comment by tseemann** [comment]
I just got the same thing on a massive collection of *S.aureus* genomes:

```
WARNING: For the k-mer size used (21), the random match probability (0.000641033) is above the specified warning threshold (0.01) for the sequence "saureus" of size 18446744072235684419. Distances to this sequence may be underestimated as a result. To meet the threshold of 0.01, a k-mer size of at least 20 is required.
```

The sequence size is ~16 Gbp of contigs.

**Comment by tseemann** [comment]
Ok, I built head with your 64 bit int patch and it now sketches them ok.

https://github.com/marbl/Mash/commit/6aad387cf37aa2362148ae84859bc2409e224565

Can you make a 1.0.2 release?

**Comment by ondovb** [comment]
Fixed in [v1.0.2](https://github.com/marbl/Mash/releases/tag/v1.0.2)
